src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -36,7 +36,7 @@ public void Add(Feature feature)
             }));
         }
 
-        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings)
+        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings, RoutingComponent routing)
```

Comment:
I am not against identifying core components and making them explicit but:
1. I don't think routing should be the first one (Receiving? Sending?)
2. I want to see at least 2 (ideally 3) of them made explicit in the same PR to be able to tell if the API is going into right direction.  Just adding stuff to this `SetupFeatures` method won't cut it I believe.
3. I am worried that if we end up with 5+ of these core components, the benefits of being explicit can become smaller than the drawbacks + benefits of treating things the same.
4. I would love e.g. to have `IDispatcher` available when setting up features